### ISSUE TITLE
Data race

### ISSUE THREAD

**Issue by ymmt2005** [topolvm/pvc-autoresizer#37]
https://github.com/topolvm/pvc-autoresizer/runs/2455739145?check_suite_focus=true

```
WARNING: DATA RACE
Read at 0x00c0005a71b0 by goroutine 132:
  runtime.chansend()
      /opt/hostedtoolcache/go/1.16.3/x64/src/runtime/chan.go:158 +0x0
  k8s.io/apimachinery/pkg/watch.(*Broadcaster).Action()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/watch/mux.go:198 +0x13a
  k8s.io/client-go/tools/record.(*recorderImpl).generateEvent.func1()
      /home/runner/go/pkg/mod/k8s.io/client-go@v0.20.2/tools/record/event.go:344 +0x88

Previous write at 0x00c0005a71b0 by goroutine 148:
  runtime.closechan()
      /opt/hostedtoolcache/go/1.16.3/x64/src/runtime/chan.go:355 +0x0
  k8s.io/apimachinery/pkg/watch.(*Broadcaster).Shutdown.func1()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/watch/mux.go:210 +0x88
  k8s.io/apimachinery/pkg/watch.(*Broadcaster).blockQueue.func1()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/watch/mux.go:109 +0x7a
  k8s.io/apimachinery/pkg/watch.(*Broadcaster).loop()
      /home/runner/go/pkg/mod/k8s.io/apimachinery@v0.20.2/pkg/watch/mux.go:221 +0x141
```

**Comment by llamerada-jp** [comment]
It seems that this error caused by the implementation of controller-runtime.
see: https://github.com/kubernetes-sigs/controller-runtime/issues/1367

We need to update controller-runtime after the above PR will be merged.

**Comment by toshipp** [comment]
As far as I know, this is resolved by #42, so close this.
